readthedocs/core/static-src/core/js/doc-embed/search.js

```diff
@@ -281,9 +281,144 @@ function attach_elastic_search_query(data) {
+                                    $('<h4>').append($('<a>', {'href': section_link, 'text': section_title}))
+                                );
+                                result.append(
+                                    $('<p>', {'text': section_content})
+                                );
+                                searchResults.append(result);
+                            }
+                        }
+                    }
+                } else {
+                    console.log('Read the Docs search returned 0 result. Falling back to MkDocs search.');
+                    fallbackSearch();
+                }
+            })
+            .fail(function (error) {
+                console.log('Read the Docs search failed. Falling back to MkDocs search.');
+                fallbackSearch();
+            });
+
+        $.ajax({
+            url: search_url.href,
+            crossDomain: true,
+            xhrFields: {
+                withCredentials: true,
+            },
+            complete: function (resp, status_code) {
+                if (
+                    status_code !== 'success' ||
+                    typeof (resp.responseJSON) === 'undefined' ||
+                    resp.responseJSON.count === 0
+                ) {
+                    return search_def.reject();
+                }
+                return search_def.resolve(resp.responseJSON);
+            }
+        })
+        .fail(function (resp, status_code, error) {
+              return search_def.reject();
+        });
+    };
+
+    var initSearch = function () {
+        var search_input = document.getElementById('mkdocs-search-query');
+        if (search_input) {
+          search_input.addEventListener('keyup', doSearch);
+        }
+
+        var term = window.getSearchTermFromLocation();
+        if (term) {
+          search_input.value = term;
+          doSearch();
+        }
+    };
+
+    $(document).ready(function () {
+        // We can't override the search completely,
+        // because we can't delete the original event listener,
+        // and MkDocs includes its search functions after ours.
+        // If MkDocs is loaded before, this will trigger a double search
+        // (but ours will have precendece).
+
+        // Note: this function is only available on Mkdocs >=1.x
+        window.doSearchFallback = window.doSearch;
+
+        window.doSearch = doSearch;
+        window.initSearch = initSearch;
+        initSearch();
+    });
+}
+
+
 function init() {
     var data = rtddata.get();
-    attach_elastic_search_query(data);
+    if (data.is_sphinx_builder()) {
+      // Check for disabled server side search for sphinx
+      // happens inside the function, because we still need to call Search.init().
+      attach_elastic_search_query_sphinx(data);
+    }
+    // MkDocs projects should have this flag explicitly for now.
+    else if (data.features && !data.features.docsearch_disabled) {
```

Comment:
Shouldn't we check for this above the sphinx check above? Otherwise we're still executing sphinx search when `docsearch_disabled` is True.

Reply:
There is an explanation above

```
        // Check to disabled server side search for Sphinx
        // happens inside the function, because we still need to call Search.init().
```